File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/ICommunityManagersView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.ArrayList;
import java.util.List;

import dev.ragnarok.fenrir.model.Manager;
import dev.ragnarok.fenrir.model.User;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface ICommunityManagersView extends IAccountDependencyView, IErrorView, IMvpView, IToastView {

    void notifyDataSetChanged();

    void displayRefreshing(boolean loadingNow);

    void displayData(List<Manager> managers);

    void goToManagerEditing(int accountId, int groupId, Manager manager);

    void showUserProfile(int accountId, User user);

    void startSelectProfilesActivity(int accountId, int groupId);

    void startAddingUsersToManagers(int accountId, int groupId, ArrayList<User> users);

    void notifyItemRemoved(int index);

    void notifyItemChanged(int index);

    void notifyItemAdded(int index);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/ICommunityManagersView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.Manager
import dev.ragnarok.fenrir.model.User
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface ICommunityManagersView : IAccountDependencyView, IErrorView, IMvpView, IToastView {
    fun notifyDataSetChanged()
    fun displayRefreshing(loadingNow: Boolean)
    fun displayData(managers: List<Manager>)
    fun goToManagerEditing(accountId: Int, groupId: Int, manager: Manager)
    fun showUserProfile(accountId: Int, user: User)
    fun startSelectProfilesActivity(accountId: Int, groupId: Int)
    fun startAddingUsersToManagers(accountId: Int, groupId: Int, users: ArrayList<User>)
    fun notifyItemRemoved(index: Int)
    fun notifyItemChanged(index: Int)
    fun notifyItemAdded(index: Int)
}